src/sentry/issues/occurrence_consumer.py

```diff
@@ -414,6 +416,16 @@ def process_occurrence_group(items: list[Mapping[str, Any]]) -> None:
     Process a group of related occurrences (all part of the same group)
     completely serially.
     """
+
+    status_changes = [
+        item for item in items if item.get("payload_type") == PayloadType.STATUS_CHANGE.value
+    ]
+
+    if len(status_changes) > 1:
+        items = [
+            item for item in items if item.get("payload_type") != PayloadType.STATUS_CHANGE.value
+        ] + status_changes[-1]
```

Comment:
processing the last status change _after_ all the occurrences in the batch;

alternatively we can keep them in order and remove all status changes but the last one.